File: solutions/3879-find-minimum-log-transportation-cost/find-minimum-log-transportation-cost.java
class Solution {
    public long minCuttingCost(int n, int m, int k) {
        long cost = 0;
        
        if (n > k) {
            int parts = n / k;
            int leftover = n - parts * k;
            cost += (long) Math.pow(k, parts) * Math.max(1, leftover);
        }

        if (m > k) {
            int parts = m / k;
            int leftover = m - parts * k;
            cost += (long) Math.pow(k, parts) * Math.max(1, leftover);
        }

        return cost;
    }
}

File: solutions/3879-find-minimum-log-transportation-cost/find-minimum-log-transportation-cost.py
class Solution:
    def minCuttingCost(self, n: int, m: int, k: int) -> int:
        cost = 0

        if n > k:
            parts = n // k
            leftover = n - parts * k
            cost += int(k ** parts) * max(1, leftover)

        if m > k:
            parts = m // k
            leftover = m - parts * k
            cost += int(k ** parts) * max(1, leftover)

        return cost